DSharpPlus/Entities/DiscordMessageBuilder.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace DSharpPlus.Entities
+{
+    /// <summary>
+    /// Constructs a Message to be sent.
+    /// </summary>
+    public sealed class DiscordMessageBuilder
+    {
+        /// <summary>
+        /// Gets the Message to be sent.
+        /// </summary>
+        public string Content { get; private set; }
+
+        /// <summary>
+        /// Gets the Embed to be sent.
+        /// </summary>
+        public DiscordEmbed Embed { get; private set; }
+
+        /// <summary>
+        /// Gets if the message should be TTS.
+        /// </summary>
+        public bool IsTTS { get; private set; } = false;
+
+        /// <summary>
+        /// Gets the Allowed Mentions for the message to be sent. 
+        /// </summary>
+        public IEnumerable<IMention> Mentions { get; private set; }
+
+        /// <summary>
+        /// Gets the Files to be sent in the Message.
+        /// </summary>
+        public Dictionary<string, Stream> Files { get; private set; } = new Dictionary<string, Stream>();
+
+        /// <summary>
+        /// Sets the Content of the Message.
+        /// </summary>
+        /// <param name="content">The content to be set.</param>
+        /// <returns></returns>
+        public DiscordMessageBuilder WithContent(string content)
+        {
+            this.Content = content;
+            return this;
+        } 
+
+        /// <summary>
+        /// Sets if the message should be TTS.
+        /// </summary>
+        /// <param name="isTTS">If TTS should be set.</param>
+        /// <returns></returns>
+        public DiscordMessageBuilder HasTTS(bool isTTS)
+        {
+            this.IsTTS = isTTS;
+            return this;
+        }
+
+        /// <summary>
+        /// Sets if the message will have an Embed.
+        /// </summary>
+        /// <param name="embed">The embed that should be sent.</param>
+        /// <returns></returns>
+        public DiscordMessageBuilder WithEmbed(DiscordEmbed embed)
+        {
+            this.Embed = embed;
+            return this;
+        }
+
+        /// <summary>
+        /// Sets if the message has allowed mentions.
+        /// </summary>
+        /// <param name="allowedMentions">The allowed Mentions that should be sent.</param>
+        /// <returns></returns>
+        public DiscordMessageBuilder HasAllowedMentions(IEnumerable<IMention> allowedMentions)
```

Comment:
Should make it consistent: `WithAllowedMentions`

Reply:
I like that idea, ill make the adjustments tonight